action/delete_vm.go

```diff
@@ -18,12 +18,7 @@ func NewDeleteVM(
 }
 
 func (a DeleteVMAction) Run(vmCID VMCID) (interface{}, error) {
-	vm, found, err := a.vmFinder.Find(int(vmCID))
-
-	if err != nil {
-		return nil, bosherr.WrapErrorf(err, "Finding vm '%s'", vmCID)
-	}
-
+	vm, found, _ := a.vmFinder.Find(int(vmCID))
```

Comment:
CPI should allow VMNotFoundError otherwise it will block the deployment. 
The action is compliance with the specification of CPI API v1
http://bosh.io/docs/cpi-api-v1.html